pkg/kudoctl/util/repo/repo_client.go

```diff
@@ -0,0 +1,69 @@
+package repo
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/kudobuilder/kudo/pkg/version"
+)
+
+//Client is the default HTTP(/S) backend handler
+type Client struct {
+	client   *http.Client
+	username string
+	password string
+}
+
+//SetCredentials sets the credentials for the RepoClient
+func (c *Client) SetCredentials(username, password string) {
+	c.username = username
+	c.password = password
+}
+
+//Get performs a Get from repo.Getter and returns the body.
+func (c *Client) Get(href string) (*bytes.Buffer, error) {
+	return c.get(href)
```

Comment:
🤔 why are we just calling private method with the same name? Why is the code not directly here?

Reply:
Typically the pattern is calling a private request function, so this can be inlined until there's other public methods.